metricflow/engine/metricflow_engine.py

```diff
@@ -244,25 +250,39 @@ class MetricFlowEngine(AbstractMetricFlowEngine):
 
     _LOGGING_INDENT = "   "
 
+    @staticmethod
+    def from_config(handler: YamlFileHandler) -> MetricFlowEngine:
+        """Initialize MetricFlowEngine via yaml config file."""
+        sql_client = make_sql_client_from_config(handler)
+        semantic_model = SemanticModel(build_user_configured_model_from_config(handler))
+        system_schema = handler.get_value(CONFIG_DWH_SCHEMA)
+        return MetricFlowEngine(
+            semantic_model=semantic_model,
+            sql_client=sql_client,
+            system_schema=system_schema,
+        )
+
     def __init__(  # noqa: D
         self,
         semantic_model: SemanticModel,
         sql_client: SqlClient,
-        column_association_resolver: ColumnAssociationResolver,
-        time_source: TimeSource,
-        time_spine_source: TimeSpineSource,
         system_schema: str,
+        column_association_resolver: Optional[ColumnAssociationResolver] = None,
+        time_source: Optional[TimeSource] = None,
+        time_spine_source: Optional[TimeSpineSource] = None,
```

Comment:
Hmmm. Looks like we only use these overrides in testing. I'd prefer not to have them in the core initializer, but I can't come up with a nice way of handling the test overrides.

I think it's worth a docstring with param annotations on the initializer here as a short term hint to people to avoid direct calls to `MetricFlowEngine()` itself.